internal/gensupport/buffer.go

```diff
@@ -28,6 +28,11 @@ func NewMediaBuffer(media io.Reader, chunkSize int) *MediaBuffer {
 	return &MediaBuffer{media: media, chunk: make([]byte, 0, chunkSize)}
 }
 
+// NewMediaBuffer initializes a MediaBuffer.
+func NewMediaBufferWithBuffer(media io.Reader, chunkSize int, buffer []byte) *MediaBuffer {
+	return &MediaBuffer{media: media, chunk: buffer}
```

Comment:
There should be a fallback here to call `NewMediaBuffer` if chunksize > buffer.

Reply:
Also, before the buffer gets passed in we should reset its length. I believe the impl expects the length to be 0.

Reply:
good idea, i had to reset buffer in caller code in order to make uploads work. Added.